redash/models/__init__.py

```diff
@@ -639,6 +647,26 @@ def all_groups_for_query_ids(cls, query_ids):
                    WHERE queries.id in :ids"""
 
         return db.session.execute(query, {'ids': tuple(query_ids)}).fetchall()
+    
+    @classmethod
+    def update_latest_result(cls, query_result):
+        # TODO: Investigate how big an impact this select-before-update makes.
+        queries = Query.query.filter(
+            Query.query_hash == query_result.query_hash,
+            Query.data_source == query_result.data_source
+        )
```

Comment:
In case there are many similar queries (e.g. forked but never changed) it may make sense to use `with_for_update` here to generate a `FOR UPDATE` SQL statement and then use `update()` here instead of iterating over the list of queries to update the query_result.

Reply:
So far this wasn't an issue, but anyway it's the same behavior we had until now -- I just moved it. So changing it is out of scope for this PR.

Reply:
Makes sense.